src/vsphere_cpi/lib/cloud/vsphere/resources/network.rb

```diff
@@ -0,0 +1,98 @@
+require 'cloud/vsphere/resources'
+
+module VSphereCloud
+  module Resources
+    class Network
+      include VimSdk
+      include ObjectStringifier
+      stringify_with :name, :mob, :bosh_name
+
+      attr_reader :mob, :client, :bosh_name
+      def self.make_network_resource(bosh_name, mob, client)
+        case mob
+        when VimSdk::Vim::Dvs::DistributedVirtualPortgroup
+          if mob.config.respond_to?(:backing_type) && mob.config.backing_type == 'nsx'
+            DistributedVirtualPortGroupNSXTNetwork.new(bosh_name, mob, client)
+          else
+            DistributedVirtualPortGroupNetwork.new(bosh_name, mob, client)
+          end
+        when VimSdk::Vim::OpaqueNetwork
+          OpaqueNetwork.new(bosh_name, mob, client)
+        else
+          Network.new(bosh_name, mob, client)
+        end
+      end
+
+      def initialize(bosh_name, mob, client)
+        @bosh_name = bosh_name
+        @mob = mob
+        @client = client
+      end
+
+      def vim_name
+        @mob.name
+      end
+
+      def nic_backing(_)
+        backing_info = VimSdk::Vim::Vm::Device::VirtualEthernetCard::NetworkBackingInfo.new
+        backing_info.device_name = vim_name
+        backing_info.network = mob
+        backing_info
+      end
+
+      def inspect
+        "<network: / #{@bosh_name} / #{@mob}"
+      end
+    end
+
+    class OpaqueNetwork < Network
+
+      def nic_backing(dvs_index)
+        backing_info = VimSdk::Vim::Vm::Device::VirtualEthernetCard::OpaqueNetworkBackingInfo.new
+        network_id = mob.summary.opaque_network_id
+        backing_info.opaque_network_id = network_id
+        dvs_index[network_id] = bosh_name
+        backing_info.opaque_network_type = mob.summary.opaque_network_type
+        backing_info
+      end
+    end
+
+    class DistributedVirtualPortGroupNetwork < Network
+
+      def nic_backing(dvs_index)
+        portgroup_properties = client.cloud_searcher.get_properties(mob,
+                                                             VimSdk::Vim::Dvs::DistributedVirtualPortgroup,
```

Comment:
Dedent this rather than aligning with the `(`. This is super hard to read.

Reply:
Done